streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -505,7 +506,7 @@ private void setupTask(final StreamsConfig streamsConfig,
                 streamsMetrics
             );
 
-            final InternalProcessorContext context = new ProcessorContextImpl(
```

Comment:
I don't think we need this change. The new test you added below, seem to re-test `ProcessorContextImpl` that is already tested in `ProcessorContextImplTest` and thus we don't need those.